Signal/src/ViewControllers/HomeView/HomeViewCell.m

```diff
@@ -111,15 +118,25 @@ - (void)commontInit
     self.snippetLabel.font = [self snippetFont];
     self.snippetLabel.numberOfLines = 1;
     self.snippetLabel.lineBreakMode = NSLineBreakByTruncatingTail;
-    [self.snippetLabel setContentHuggingHorizontalLow];
-    [self.snippetLabel setCompressionResistanceHorizontalLow];
 
-    UIStackView *bottomRowView = [[UIStackView alloc] initWithArrangedSubviews:@[
+    self.typingIndicatorView = [TypingIndicatorView new];
+
+    self.previewStackView = [[UIStackView alloc] initWithArrangedSubviews:@[
         self.snippetLabel,
+        self.typingIndicatorView,
+    ]];
+    self.previewStackView.axis = UILayoutConstraintAxisVertical;
+    self.previewStackView.alignment = UIStackViewAlignmentLeading;
+    [self.previewStackView setContentHuggingHorizontalLow];
+    [self.previewStackView setCompressionResistanceHorizontalLow];
```

Comment:
The typing indicator replaces the "snippet" we use to preview the conversation contents.  The only tricky thing was ensuring that the cell had a consistent layout between these two cases.

Reply:
stacks on stacks on stacks on stacks. Let's make sure we verify this layout on legacy iOS versions as part of the test plan, since iOS9/10 can behave somewhat differently with complex stack views.

Reply:
Noted.